modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/DateTimeObjectFieldBusinessType.java

```diff
@@ -0,0 +1,204 @@
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.DateUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.vulcan.extension.PropertyDefinition;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Murilo Stodolni
+ */
+@Component(
+	property = "object.field.business.type.key=" + ObjectFieldConstants.BUSINESS_TYPE_DATE_TIME,
+	service = ObjectFieldBusinessType.class
+)
+public class DateTimeObjectFieldBusinessType
+	implements ObjectFieldBusinessType {
+
+	@Override
+	public Set<String> getAllowedObjectFieldSettingsNames() {
+		return Collections.singleton(
+			ObjectFieldSettingConstants.NAME_TIME_STORAGE);
+	}
+
+	@Override
+	public String getDBType() {
+		return ObjectFieldConstants.DB_TYPE_DATE_TIME;
+	}
+
+	@Override
+	public String getDDMFormFieldTypeName() {
+		return DDMFormFieldTypeConstants.DATE_TIME;
+	}
+
+	@Override
+	public String getDescription(Locale locale) {
+		return _language.get(locale, "add-date-and-time-value");
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "date-and-time");
+	}
+
+	@Override
+	public String getName() {
+		return ObjectFieldConstants.BUSINESS_TYPE_DATE_TIME;
+	}
+
+	@Override
+	public PropertyDefinition.PropertyType getPropertyType() {
+		return PropertyDefinition.PropertyType.DATE_TIME;
+	}
+
+	@Override
+	public Set<String> getRequiredObjectFieldSettingsNames(
+		ObjectField objectField) {
+
+		return Collections.singleton(
+			ObjectFieldSettingConstants.NAME_TIME_STORAGE);
+	}
+
+	@Override
+	public Set<String> getUnmodifiableObjectFieldSettingsNames() {
+		return Collections.singleton(
+			ObjectFieldSettingConstants.NAME_TIME_STORAGE);
+	}
+
+	@Override
+	public Object getValue(
+			ObjectField objectField, long userId, Map<String, Object> values)
+		throws PortalException {
+
+		if (!values.containsKey(objectField.getName())) {
+			return null;
+		}
+
+		String value = String.valueOf(values.get(objectField.getName()));
+
+		if (Validator.isNull(value)) {
+			return null;
+		}
+
+		return _getLocalDateTime(
+			ObjectFieldSettingUtil.getTimeZoneId(
+				objectField.getObjectFieldSettings(),
+				_userLocalService.getUser(userId)),
+			StringPool.UTC, value);
+	}
+
+	@Override
+	public Object getValueToDisplayContext(
+			ObjectField objectField, long userId, Map<String, Object> values)
+		throws PortalException {
+
+		String value = String.valueOf(values.get(objectField.getName()));
+
+		if (Validator.isNull(value)) {
```

Comment:
The way I see it, the `if (!values.containsKey(objectField.getName())) {`, actually looks redundant, since we're already calling `if (Validator.isNull(value)) {`. And if the `values` map does not contain the `objectField.getName()` it will fall there.

Reply:
@devpedrotavares agree with you.. please remove from the other method as it also seems redundant so we can keep the pattern

Reply:
I just saw that you already did this. Thanks!